### PR TITLE
added concat operation

### PR DESCRIPTION
**Pull request by bergos**
This should cover the [sequence](https://github.com/zazuko/blv-tierseuchen-ld/blob/master/lib/sequence.js) code + other readable streams.

### CODE REVIEW

File: lib/concat.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+}
+
+function factory (...streams) {
+  return new ConcatStream(streams)
+}
+
+factory.object = (...streams) => {
+  return new ConcatStream(streams, { objectMode: true })
+}
```
Comment (tpluscode) [inline lib/concat.js:44]: In turtle, this would be referenced with dot notation as `node:barnard59-base/concat.object`?

**Review by tpluscode — COMMENTED** [review]
The `sequence.js` from BLV specifically operates on pipelines. IIRC I called `resume()` specifically to do that.

Will this `concat` allow merging of any stream, including pipelines?

**Comment by bergos** [comment]
I tested it manually in a pipeline. The automatic tests uses plain `Readable` streams.

File: lib/concat.js (hunk 1 above)
Comment (bergos) [reply lib/concat.js:44]: Jup. This is already used in `json` and `fetch`.

**Review by tpluscode — APPROVED** [review]

**Pull request merged**